Content/default/webpack.config.js

```diff
@@ -28,113 +17,37 @@ var CONFIG = {
     devServerProxy: {
         // redirect requests that start with /api/ to the server on port 8085
         '/api/**': {
-            target: 'http://localhost:' + (process.env.SERVER_PROXY_PORT || "8085"),
+            target: 'http://localhost:' + (process.env.SERVER_PROXY_PORT || '8085'),
             changeOrigin: true
         },
         // redirect websocket requests that start with /socket/ to the server on the port 8085
         '/socket/**': {
-            target: 'http://localhost:' + (process.env.SERVER_PROXY_PORT || "8085"),
+            target: 'http://localhost:' + (process.env.SERVER_PROXY_PORT || '8085'),
             ws: true
         }
     }
 }
 
-console.log('Bundling for ' + mode + '...');
+const TEST_CONFIG = {
+    // The tags to include the generated JS and CSS will be automatically injected in the HTML template
+    // See https://github.com/jantimon/html-webpack-plugin
+    indexHtmlTemplate: 'tests/Client/index.html',
+    fsharpEntry: 'tests/Client/output/Client.Tests.js',
+    outputDir: 'tests/Client',
+    assetsDir: 'tests/Client',
+    devServerPort: 8081,
+    // When using webpack-dev-server, you may need to redirect some calls
+    // to a external API server. See https://webpack.js.org/configuration/dev-server/#devserver-proxy
+    devServerProxy: undefined,
+}
 
-// The HtmlWebpackPlugin allows us to use a template for the index.html page
-// and automatically injects <script> or <link> tags for generated bundles.
-var commonPlugins = [
-    new HtmlWebpackPlugin({
-        filename: 'index.html',
-        template: resolve(CONFIG.indexHtmlTemplate)
-    })
-];
+module.exports = function(env, arg) {
+    // Mode is passed as a flag to npm run. see the docs for more details on flags https://webpack.js.org/api/cli/#flags
+    const mode = arg.mode ?? 'development';
+    // environment variables docs: https://webpack.js.org/api/cli/#environment-options
+    const config = env.test ? TEST_CONFIG : CONFIG;
```

Comment:
If you have those two configurations in same file, why the extra `webpack.common.js` file? Would be simpler to just have everything inlined into single file right?

Reply:
Yeah, I was thinking about it as well. I was not sure which one would look cleaner. 
Also ehere would you place the build function, I was thinking two options:
Option 1:
```javascript
const CONFIG = { /*...*/}
const TEST_CONFIG = {/*...*/}
module.exports = function(env, arg) { /*...*/ }

//build function at the bottom
function buildConfig(config, mode) { /*...*/ }
```

Option 2:
```javascript
const CONFIG = { /*...*/}
const TEST_CONFIG = {/*...*/}
function buildConfig(config, mode) { /*...*/ }
// module.exports as last statement
module.exports = function(env, arg) { /*...*/ }
```

any preference?

Reply:
I pushed a commit with Option 1. Please let me know if you have any feedback.

Reply:
If you're asking, I think option 2. Or just not have a `buildConfig` function at all and just do it all inline.

Reply:
I just pushed a commit following your suggestion.
To be honest I don't have a particular preference, all these options look good to me. But I'm happy to try alternatives until we find one that looks good for all of us.